src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1775,13 +1800,37 @@
   (contains? (ks/keyset user-name->query-rec-name)
              (first expr)))
 
+(defn create-json-subtree-projection
+  [dotted-field projections]
+  (let [json-path (str/split dotted-field #"\.")
+        stringify-field (fn [{:keys [field field-type]}]
+                          (case field-type
+                            :keyword (name field)
+                            :raw (:s field)
+                            (throw (IllegalArgumentException.
+                                           (tru "Cannot determine field type for field ''{0}''" field)))))]
+  {:type :json
+   :queryable? false
+   :field (hcore/raw
+            (str
+              ; Extract the original field as a string
+              (->> json-path
+                   first
+                   projections
+                   stringify-field)
+              ; append any required ->'key' operators
+              (when (> (count json-path) 1)
```

Comment:
Could say `(pos? (count json-path))`, but think it's more idomatic to say `(seq json-path)`.

Reply:
I think that would only work for `> 0`, here I need to check that  there are actually json keys to look up.